### PR TITLE
Wire in Confirmation Dialog

### PR DESCRIPTION
**Pull request by nwmac**
Tidied up tabs base to remove edit stuff not used
Wired in dialog to most actions in the application view:
 - Routes
 - Instances
 - Start/Stop/Delete actions

### CODE REVIEW

File: src/app/features/applications/application/application-tabs-base/application-tabs-base.component.ts [hunk 1]
```diff
@@ -136,37 +114,8 @@ export class ApplicationTabsBaseComponent implements OnInit, OnDestroy {
     ).map(([isFetchingApp, isFetchingEnvVars, isFetchingStats]) => {
       return isFetchingApp || isFetchingEnvVars || isFetchingStats;
     }).distinctUntilChanged();
-
-    this.summaryDataChanging$ = Observable.combineLatest(
```
Comment (richard-cox) [inline src/app/features/applications/application/application-tabs-base/application-tabs-base.component.ts]: This is still referenced in a few places in application-tabs-base.component.html

File: src/app/features/applications/application/application-tabs-base/application-tabs-base.component.ts [hunk 2]
```diff
@@ -136,37 +114,8 @@ export class ApplicationTabsBaseComponent implements OnInit, OnDestroy {
...
-          memory: application.app.entity.memory,
-          enable_ssh: application.app.entity.enable_ssh,
-          environment_json: application.app.entity.environment_json
-        };
-      }));
   }
 
-
   ngOnDestroy() {
     this.sub.forEach(subscription => subscription.unsubscribe());
```
Comment (richard-cox) [inline src/app/features/applications/application/application-tabs-base/application-tabs-base.component.ts]: Doesn't look like we're using this array anymore

File: src/app/shared/components/dialog-confirm/dialog-confirm.component.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<h1 mat-dialog-title>{{data.title}}</h1>
+<div mat-dialog-content>
+  <p>{{ data.msg }}</p>
+</div>
+<div mat-dialog-actions class="confirm-dialog__actions">
+  <button mat-button (click)="onNoClick()">Cancel</button>
+  <button mat-button [mat-dialog-close]="true" cdkFocusInitial>{{ data.confirm }}</button>
```
Comment (richard-cox) [inline src/app/shared/components/dialog-confirm/dialog-confirm.component.html]: In the steppers we add color attributes to the buttons
- previous `color="accent"` 
- cancel `color="warn"` 
- positive `color="primary"`

Not sure about the cancel, but we should probably add the positive to confirm

Comment (nwmac) [reply src/app/shared/components/dialog-confirm/dialog-confirm.component.html]: According to Angular Material:

https://material.io/guidelines/components/dialogs.html#dialogs-alerts 
https://material.io/guidelines/patterns/confirmation-acknowledgement.html

.. we should not!

They do have button text in color and the text color is primary - but not decorated as we do in the stepper.

**Pull request merged**